sdk/storage/Azure.Storage.Common/src/Shared/StorageServerTimeoutPolicy.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Storage
+{
+    internal class StorageServerTimeoutPolicy : HttpPipelineSynchronousPolicy
+    {
+        private StorageServerTimeoutPolicy()
+        {
+        }
+
+        public static StorageServerTimeoutPolicy Shared { get; } = new StorageServerTimeoutPolicy();
+
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            if (message.TryGetProperty(Constants.ServerTimeout.HttpMessagePropertyKey, out var value))
+            {
+                if (value is int timeout)
+                {
+                    string query = message.Request.Uri.Query;
+                    if (string.IsNullOrEmpty(query))
+                    {
+                        message.Request.Uri.Query += $"?{Constants.ServerTimeout.QueryParameterKey}={timeout}";
```

Comment:
String interpolation is fine for constants, but you don't want to include parameters like this without specifying the culture/formatting.